parquet/src/main/java/org/apache/iceberg/parquet/ParquetBloomRowGroupFilter.java

```diff
@@ -0,0 +1,259 @@
+      // bloom filter is based on hash and cannot eliminate based on lt or ltEq or gt or gtEq
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean gtEq(BoundReference<T> ref, Literal<T> lit) {
+      // bloom filter is based on hash and cannot eliminate based on lt or ltEq or gt or gtEq
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
+      int id = ref.fieldId();
+
+      Boolean hasNonBloomFilter = isFallback.get(id);
+      if (hasNonBloomFilter == null || hasNonBloomFilter) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      BloomFilter bloom = getBloomById(id);
+      T value = lit.value();
+      return bloom.findHash(tryHash(id, value, bloom)) ? ROWS_MIGHT_MATCH : ROWS_CANNOT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean notEq(BoundReference<T> ref, Literal<T> lit) {
+      // bloom filter is based on hash and cannot eliminate based on notEq
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean in(BoundReference<T> ref, Set<T> literalSet) {
+      int id = ref.fieldId();
+
+      Boolean hasNonBloomFilter = isFallback.get(id);
+      if (hasNonBloomFilter == null || hasNonBloomFilter) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      BloomFilter bloom = getBloomById(id);
+      for (T e : literalSet) {
+        if (bloom.findHash(tryHash(id, e, bloom))) {
+          // found hash so rows match
+          return ROWS_MIGHT_MATCH;
+        }
+      }
+
+      // no hash found so rows don't match
+      return ROWS_CANNOT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean notIn(BoundReference<T> ref, Set<T> literalSet) {
+      // bloom filter is based on hash and cannot eliminate based on notIn
+      return ROWS_MIGHT_MATCH;
+    }
+
+    @Override
+    public <T> Boolean startsWith(BoundReference<T> ref, Literal<T> lit) {
+      // bloom filter is based on hash and cannot eliminate based on startsWith
+      return ROWS_MIGHT_MATCH;
+    }
+
+    private BloomFilter getBloomById(int id) {
+      ColumnChunkMetaData columnChunkMetaData = columnMetaMap.get(id);
+      BloomFilter bloomFilter = bloomReader.readBloomFilter(columnChunkMetaData);
+
+      if (bloomFilter == null) {
+        throw new IllegalStateException("Failed to read required bloom filter for id: " + id);
+      }
+
+      return bloomFilter;
+    }
+
+    private <T> long tryHash(int id, T value, BloomFilter bloom) {
+      ColumnDescriptor col = cols.get(id);
+
+      switch (col.getPrimitiveType().getPrimitiveTypeName()) {
+        case BINARY:
+          return bloom.hash(Binary.fromString(value.toString()));
+        case INT32:
+        case INT64:
+        case FLOAT:
+        case DOUBLE:
+          return bloom.hash(value);
+        default:
+          throw new IllegalArgumentException("Cannot hash of type: " + col.getPrimitiveType().getPrimitiveTypeName());
```

Comment:
In case BINARY, we ran into "Not supported type" error with [BlockSplitBloomFilter.hash()](https://github.com/apache/parquet-mr/blob/apache-parquet-1.12.0/parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BlockSplitBloomFilter.java#L312) because String is not of type Binary.
This is an ugly solution and it only considers String type under BINARY.
Maybe there needs to be a function to transfer Iceberg type to Parquet type, like the reverse of ParquetConversions.converterFromParquet(colType, icebergType)?